c/extras/py_ftrl.cc

```diff
@@ -23,127 +23,220 @@
+            << "Ftrl constructor, but not both at the same time";
+    }
+  } else {
+    if (defined_alpha) {
+      fp.alpha = args[1].to_double();
+    }
 
-  if (!(args[1].is_undefined() || args[1].is_none())) {
-    fmp.b = args[1].to_double();
-  }
+    if (defined_beta) {
+      fp.beta = args[2].to_double();
+    }
 
-  if (!(args[2].is_undefined() || args[2].is_none())) {
-    fmp.l1 = args[2].to_double();
-  }
+    if (defined_lambda1) {
+      fp.lambda1 = args[3].to_double();
+    }
 
-  if (!(args[3].is_undefined() || args[3].is_none())) {
-    fmp.l2 = args[3].to_double();
-  }
+    if (defined_lambda2) {
+      fp.lambda2 = args[4].to_double();
+    }
 
-  if (!(args[4].is_undefined() || args[4].is_none())) {
-    fmp.d = static_cast<uint64_t>(args[4].to_size_t());
-  }
+    if (defined_d) {
+      fp.d = static_cast<uint64_t>(args[5].to_size_t());
+    }
 
-  if (!(args[5].is_undefined() || args[5].is_none())) {
-    fmp.n_epochs = args[5].to_size_t();
-  }
+    if (defined_n_epochs) {
+      fp.n_epochs = args[6].to_size_t();
+    }
 
-  if (!(args[6].is_undefined() || args[6].is_none())) {
-    fmp.inter = args[6].to_bool_strict();
-  }
+    if (defined_inter) {
+      fp.inter = args[7].to_bool_strict();
+    }
 
-  if (!(args[7].is_undefined() || args[7].is_none())) {
-    fmp.hash_type = static_cast<unsigned int>(args[7].to_size_t());
-  }
+    if (!(args[8].is_undefined() || args[8].is_none())) {
+      fp.hash_type = static_cast<unsigned int>(args[8].to_size_t());
+    }
 
-  if (!(args[8].is_undefined() || args[8].is_none())) {
-    fmp.seed = static_cast<unsigned int>(args[8].to_size_t());
+    if (!(args[9].is_undefined() || args[9].is_none())) {
+      fp.seed = static_cast<unsigned int>(args[9].to_size_t());
+    }
   }
 
-  fm = new FtrlModel(fmp);
+  ft = new Ftrl(fp);
 }
 
 
-void Ftrl::m__dealloc__() {
-  delete fm;
+void PyFtrl::m__dealloc__() {
+  delete ft;
 }
 
 
-const char* Ftrl::Type::classname() {
+const char* PyFtrl::Type::classname() {
   return "datatable.core.Ftrl";
 }
 
 
-const char* Ftrl::Type::classdoc() {
+const char* PyFtrl::Type::classdoc() {
   return R"(Follow the Regularized Leader (FTRL) model with hashing trick.
     
 See this reference for more details:
 https://www.eecs.tufts.edu/~dsculley/papers/ad-click-prediction.pdf
 
 Parameters
 ----------
-a : float
+alpha : float
     `alpha` in per-coordinate learning rate formula.
-b : float
+beta : float
     `beta` in per-coordinate learning rate formula.
-l1 : float
+lambda1 : float
     L1 regularization parameter.
-l2 : float
+lambda2 : float
     L2 regularization parameter.
 d : int
     Number of bins to be used after the hashing trick. 
 n_epochs : int
     Number of epochs to train for.
 inter : bool
     If feature interactions to be used or not.
-hash_type : int
-    Hashing method to use for strings:
-    `0` - std::hash;
-    `1` - Murmur2;
-    `2` - Murmur3.
-seed: int
-    Seed to be used for Murmur hash functions.
```

Comment:
Why were these removed here, yet they are still parsed in the constructor?

Reply:
Yes, I don't think we will need them after working on #1391. I will clean the rest of the code, because these parameters don't have any effect on the current hashers in any case.